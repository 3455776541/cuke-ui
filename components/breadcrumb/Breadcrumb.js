import React, { PureComponent, cloneElement } from 'react';
import cls from 'classnames';
import PropTypes from 'prop-types';

export default class Breadcrumb extends PureComponent {
	static defaultProps = {
		prefix: 'cuke-breadcrumb',
		separator: '/'
	};

	static propTypes = {
		separator: PropTypes.oneOfType([
			PropTypes.string.isRequired,
			PropTypes.object.isRequired,
		])
	};

	render() {
		const { prefix, className, separator, children, ...attr } = this.props;

		const items = React.Children.map(children, (element, index) => {
			return cloneElement(element, {
				separator,
				key: index
			});
		});

		return (
			<div
				className={cls(
					prefix,
					className
				)}
				{...attr}
			>
				{items}
			</div>
		);
	}
}
