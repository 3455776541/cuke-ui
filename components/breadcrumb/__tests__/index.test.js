import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Breadcrumb from "../index";

describe("<Breadcrumb/>", () => {
  it("should render Breadcrumb", () => {
    const wrapper = render(
      <Breadcrumb>
        <Breadcrumb.Item>首页</Breadcrumb.Item>
        <Breadcrumb.Item>模块</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render custom separator", () => {
    const wrapper = render(
      <Breadcrumb separator=">">
        <Breadcrumb.Item>黄瓜 ui</Breadcrumb.Item>
        <Breadcrumb.Item>面包屑</Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="#">链接</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>导航</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
