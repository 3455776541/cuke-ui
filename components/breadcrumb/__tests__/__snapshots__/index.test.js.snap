// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb/> should render Breadcrumb 1`] = `
<div
  class="cuke-breadcrumb"
>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      首页
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      /
    </span>
  </span>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      模块
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      /
    </span>
  </span>
</div>
`;

exports[`<Breadcrumb/> should render custom separator 1`] = `
<div
  class="cuke-breadcrumb"
>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      黄瓜 ui
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      &gt;
    </span>
  </span>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      面包屑
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      &gt;
    </span>
  </span>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      <a
        href="#"
      >
        链接
      </a>
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      &gt;
    </span>
  </span>
  <span
    class="cuke-breadcrumb-item"
  >
    <span
      class="cuke-breadcrumb-item-text"
    >
      导航
    </span>
    <span
      class="cuke-breadcrumb-item-separator"
    >
      &gt;
    </span>
  </span>
</div>
`;
